### PR TITLE
SDIT-1125 Refactor case note count tests and remove unused /api/bookings/offenderNo/{offenderNo}/caseNotes endpoint

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceIntTest_caseNotes.kt [hunk 1]
```diff
@@ -15,7 +15,7 @@ import uk.gov.justice.hmpps.prison.repository.jpa.repository.OffenderCaseNoteRep
 import uk.gov.justice.hmpps.prison.util.DateTimeConverter
 import java.time.ZoneOffset
 
-class BookingCaseNotesResourceIntTest : ResourceTest() {
+class BookingResourceIntTest_caseNotes : ResourceTest() {
```
Comment (karenmillermoj) [inline src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceIntTest_caseNotes.kt:18]: Renamed file to match others

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/BookingResource.java [hunk 2]
```diff
@@ -571,18 +571,6 @@ public CaseNote createBookingCaseNote(@PathVariable("bookingId") @Parameter(desc
         return caseNoteService.createCaseNote(bookingId, body, authenticationFacade.getCurrentUsername());
     }
 
-    @ApiResponses({
-        @ApiResponse(responseCode = "201", description = "The Case Note has been recorded. The updated object is returned including the status.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = CaseNote.class))}),
-        @ApiResponse(responseCode = "409", description = "The case note has already been recorded under the booking. The current unmodified object (including status) is returned.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))})})
-    @Operation(summary = "Create case note for offender.", description = "Create case note for offender.", hidden = true)
-    @ResponseStatus(HttpStatus.CREATED)
-    @PostMapping("/offenderNo/{offenderNo}/caseNotes")
-    @HasWriteScope
```
Comment (karenmillermoj) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/BookingResource.java:580]: Endpoint no longer used (POST api/offenders/{offenderNo}/caseNotes used instead)

File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceIntTest_caseNotes.kt [hunk 3]
```diff
@@ -563,4 +563,163 @@ class BookingCaseNotesResourceIntTest : ResourceTest() {
     val ocn = offenderCaseNoteRepository.findById(caseNoteId).get()
     offenderCaseNoteRepository.delete(ocn)
   }
+
+  @Nested
+  @DisplayName("GET /api/bookings/{bookingId}/caseNotes/{type}/{subType}/count")
+  inner class CaseNoteCount {
+
+    @Test
+    fun `returns 200 if has override ROLE_SYSTEM_USER`() {
```
Comment (karenmillermoj) [inline src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceIntTest_caseNotes.kt:572]: This test is new and will change once the SYSTEM_USER update is completed

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**